src/core_codemods/harden_pyyaml.py

```diff
@@ -50,18 +53,70 @@ def rule(cls):
                                 - pattern: yaml.BaseLoader
                                 - pattern: yaml.FullLoader
                                 - pattern: yaml.UnsafeLoader
+                  - patterns:
+                      - pattern: |
+                          class $X(...,$LOADER, ...):
+                            ...
+                      - metavariable-pattern:
+                          metavariable: $LOADER
+                          patterns:
+                            - pattern-either:
+                                - pattern: yaml.Loader
+                                - pattern: yaml.BaseLoader
+                                - pattern: yaml.FullLoader
+                                - pattern: yaml.UnsafeLoader
 
         """
 
-    def on_result_found(self, original_node, updated_node):
-        maybe_name = self.get_aliased_prefix_name(original_node, self._module_name)
-        maybe_name = maybe_name or self._module_name
-        if maybe_name == self._module_name:
-            self.add_needed_import(self._module_name)
-        new_args = [
-            *updated_node.args[:1],
-            updated_node.args[1].with_changes(
-                value=self.parse_expression(f"{maybe_name}.SafeLoader")
-            ),
-        ]
-        return self.update_arg_target(updated_node, new_args)
+    def on_result_found(
+        self,
+        original_node: Union[cst.Call, cst.ClassDef],
+        updated_node: Union[cst.Call, cst.ClassDef],
+    ):
+        match original_node:
+            case cst.Call():
+                maybe_name = self.get_aliased_prefix_name(
+                    original_node, self._module_name
+                )
+                maybe_name = maybe_name or self._module_name
+                if maybe_name == self._module_name:
+                    self.add_needed_import(self._module_name)
+                new_args = [
+                    *updated_node.args[:1],
+                    updated_node.args[1].with_changes(
+                        value=self.parse_expression(f"{maybe_name}.SafeLoader")
+                    ),
+                ]
+                return self.update_arg_target(updated_node, new_args)
+            case cst.ClassDef():
+                return updated_node.with_changes(
+                    bases=self._update_bases(original_node)
+                )
+        return updated_node
+
+    def _update_bases(self, original_node: cst.ClassDef) -> list[cst.Arg]:
+        new = []
+        unsafe_name_matchers = (
+            matchers.Name(value="UnsafeLoader")
+            | matchers.Name(value="Loader")
+            | matchers.Name(value="BaseLoader")
+            | matchers.Name(value="FullLoader")
+        )
+        for base_arg in original_node.bases:
```

Comment:
This logic isn't quite correct. Try adding another test case with this as the input code:
```python
        from abc import ABC
        import yaml as yam
        from whatever import Loader

        class MyCustomLoader(ABC, Loader, yam.UnsafeLoader):
            def __init__(self, *args, **kwargs):
                super(MyCustomLoader, self).__init__(*args, **kwargs)
```
The `Loader` base in this case should not be replaced.